ThePensionsRegulator.Frontend.Umbraco/Services/TprBoxViewInterceptor.cs

```diff
@@ -0,0 +1,49 @@
+﻿using GovUk.Frontend.Umbraco;
+using GovUk.Frontend.Umbraco.Blocks;
+using Microsoft.Extensions.Options;
+using ThePensionsRegulator.Umbraco;
+using ThePensionsRegulator.Umbraco.Blocks;
+
+namespace ThePensionsRegulator.Frontend.Umbraco.Services
+{
+    public class TprBoxViewInterceptor : IBlockViewInterceptor
+    {
+        private readonly GovUkFrontendUmbracoOptions _options;
+
+        public TprBoxViewInterceptor(IOptions<GovUkFrontendUmbracoOptions> options)
+        {
+            _options = options.Value;
+        }
+
+        public void InterceptBlockView(BlockViewModel blockViewModel)
+        {
+            if (!_options.RenderWidthContainerForBlocks) { return; }
+
+            bool currentBlockIsFullWidthBox = BlockIsFullWidthBox(blockViewModel.CurrentBlock);
+
+            if (currentBlockIsFullWidthBox)
+            {
+                blockViewModel.OpenWidthContainer = false;
+                blockViewModel.CloseWidthContainer = false;
+            }
+
+            bool? previousBlockIsFullWidthBox = blockViewModel.PreviousBlock is not null ? BlockIsFullWidthBox(blockViewModel.PreviousBlock) : null;
```

Comment:
The `null` is representing that the first block doesn't have a previous block, and the last block doesn't have a next block.

When two consecutive blocks are determined to be the same, in terms of their row and column requirements, they are merged into a single row/column. We can't use `bool` because then we'd have `false` instead of `null`, and `null != false` but `false == false`. The first block might be seen as the same as a previous one that doesn't exist, or the last might be the same as a next block that doesn't exist, and the rendering would be wrong.